sat/cli/firmware/parser.py

```diff
@@ -53,3 +53,8 @@ def add_firmware_subparser(subparsers):
         '--snapshots', dest='snapshots', metavar='SNAPSHOT', nargs='*',
         help='Describe firmware snapshots. Provide no arguments to this '
              'option to list available snapshots.')
+
+    firmware_parser.add_argument(
```

Comment:
You might want to look into a way to make this mutually exclusive with the `--snapshots` option.

I don't really like how this command was structured to begin with, namely, I don't like that `sat firmware --snapshots` just lists existing snapshots and `sat firmware --snapshots SNAPSHOT_1 [SNAPSHOT_2 ...]` will show the firmware versions in those snapshots. I think it will be hard to change that behavior in a backwards compatible way.

It would just be nice to avoid any confusion in case somebody thinks that `sat firmware --delete-snapshot` means that it will create a snapshot, show the versions and then delete the snapshot. It might also be confusing that `sat firmware --delete-snapshot SNAPSHOT_1 --snapshots SNAPSHOT_1` will just delete SNAPSHOT_1 and will not show its contents. Perhaps you can think of a better way to structure the command line arguments.

Reply:
I made `--snapshots` and `--delete-snapshot` as mutually exclusive. So user might not get confused here as they can use only one at a time. So  `sat firmware --delete-snapshot SNAPSHOT_1 --snapshots SNAPSHOT_1` would be invalid.

To be more specific `sat firmware --delete-snapshot SNAPSHOT_1` deletes the snapshot SNAPSHOT_1 if it is valid snapshot-name else throws an error.